qp_klp/Amplicon.py

```diff
@@ -93,15 +93,22 @@ def generate_reports(self):
 
     def _get_data_type(self, prep_file_path):
         metadata = Step.parse_prep_file(prep_file_path)
-        if 'target_gene' in metadata.columns:
+        target_genes = []
+        for sample_name in metadata:
+            row = metadata[sample_name]
+            if 'target_gene' in row:
+                target_genes.append(row['target_gene'])
```

Comment:
Why not use pandas methods for this?
```
if 'target_gene' in metadata.columns:
   target_genes = metadata.target_gene.unique()
else:
   target_gene = []

Reply:
It's because metadata is a dict(), not a DataFrame. I'll include an option to return a DataFrame instead and modify accordingly.